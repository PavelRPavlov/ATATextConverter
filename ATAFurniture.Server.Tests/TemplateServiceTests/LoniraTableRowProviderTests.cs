using FluentAssertions;
using Kroiko.Domain.CellsExtracting;
using Kroiko.Domain.TemplateBuilding.Lonira;
using Snapshooter.Xunit;
using Xunit;

namespace ATAFurniture.Server.Tests.TemplateServiceTests;

public class LoniraTableRowProviderTests
{
    [Fact]
    public async Task GivenDetailWithFullEdges_WhenCreatingTableRows_ThenReturnsCorrectRows()
    {
        var provider = new LoniraTableRowProvider();
        var detail = new Detail(50, 100, 2, "Material", false, true, true, true, true, "Cabinet", 1, 18,1,0,0,0);
        
        var result = provider.GetTableRow(detail, 1, 1);
        
        result.Should().MatchSnapshot();
    }
    
    [Fact]
    public async Task GivenDetailWithOneLongEdge_WhenCreatingTableRows_ThenReturnsCorrectRows()
    {
        var provider = new LoniraTableRowProvider();
        var detail = new Detail(50, 100, 2, "Material", false, false, true, false, false, "Cabinet", 1, 18,1,0,0,0);
        
        var result = provider.GetTableRow(detail, 1, 1);
        
        result.Should().MatchSnapshot();
    }
    
    [Fact]
    public async Task GivenDetailWithOneLongEdgeAndRotatedGrain_WhenCreatingTableRows_ThenReturnsCorrectRows()
    {
        var provider = new LoniraTableRowProvider();
        var detail = new Detail(50, 100, 2, "Material", true, false, true, false, false, "Cabinet", 1, 18,1,0,0,0);
        
        var result = provider.GetTableRow(detail, 1, 1);
        
        result.Should().MatchSnapshot();
    }
}