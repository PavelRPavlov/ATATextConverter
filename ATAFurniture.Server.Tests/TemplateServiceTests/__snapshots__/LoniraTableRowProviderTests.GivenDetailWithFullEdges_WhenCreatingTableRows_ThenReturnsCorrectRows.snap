[
  {
    "Name": "A1",
    "ContentAlignment": 1,
    "Value": "50",
    "Row": 1
  },
  {
    "Name": "B1",
    "ContentAlignment": 1,
    "Value": "100",
    "Row": 1
  },
  {
    "Name": "C1",
    "ContentAlignment": 1,
    "Value": "2",
    "Row": 1
  },
  {
    "Name": "D1",
    "ContentAlignment": 0,
    "Value": "2 k 2 d; Cabinet",
    "Row": 1
  }
]
